libs/libvtrutil/src/vtr_vector_map.h

```diff
@@ -62,7 +62,7 @@ class vector_map {
     //Indexing
     const_reference operator[](const K n) const {
         size_t index = size_t(n);
```

Comment:
I'm nervous about losing the check.  I'd move the VTR_ASSERT_SAFE check for index>=0 to before the size_t conversion (i.e.
VTR_ASSERT_SAFE (n>=0)
would come right after the function header.

Reply:
I would agree, but vector_map is only used by StrongId types, which only allow conversion to size_t, thus will never be negative. If a negative number got involved, it would wrap around and be caught by the `index < vec_.size()` check. This could result in undefined behaviour, but this would have to be handled by the StrongId type, not vector_map.

Reply:
I tried adding the ASSERT_SAFE (n >= 0) to the top, but it won't compile since StrongId doesn't have a < operator.

Reply:
OK, sounds good.  Adding @kmurray in case he has any objections, but that sounds reasonable.

Reply:
David, can you add a comment explaining this (we're converting to size_t, which can't be negative, so just check if it is too big).

Reply:
Yeah sure that's not a bad idea. Will do!